### ISSUE TITLE
Line-spacing on survey pattern creation.

### ISSUE THREAD

**Issue by valschmidt** [rolker/AutonomousMissionPlanner#40]
I showed JB our survey pattern tool and he suggested that, in addition to dynamically showing the total distance and time as one manipulates the widgets, we also show the line spacing. I think it's a good idea.

I have some other ideas. I have a matlab script in which I specify the parameters of a survey area (mean water depth, length, width), sonar parameters (swath half angle, number of soundings per ping, number of swaths per ping cycle, equidistant vs equiangular) and some desired survey parameters (percent overlap, alteration fraction which changes the line spacing on alternating lines, desired survey speed and maximum survey time). The script calculates and displays in a plot the lines for the survey (which may or may not cover more the whole specified survey area), shows and calculates the area coverage in square km, estimates the time to complete with some assumptions about the time for turns, and calculates the resulting data density (soundings/m^2).   Here's an example where the original survey area was 10km x 5km, the speed was 5.5 knots, the survey time was 10 hours:

![image](https://user-images.githubusercontent.com/2421975/53174272-71708e80-35b7-11e9-8f63-1e42ac116020.png)

It would be useful to do the same dynamically as we draw a survey. We could specify all these parameters ahead of time, and then maybe provide dynamic guides/widgets which indicate things like minimum estimated sounding density, when the survey can be completed in the time allotted, etc.

**Comment by rolker** [comment]
Yes, I like idea of implementing a dynamic version of your script.
Using a mean water depth certainly simplifies a few things instead of reading depths from a chart or existing survey.
